flink-runtime-web/web-dashboard/src/app/app.component.html

```diff
@@ -49,6 +49,9 @@ <h1>Apache Flink Dashboard</h1></a>
         <span><i nz-icon type="upload"></i><span>Submit New Job</span></span>
       </li>
     </ul>
+    <nz-affix *ngIf="!collapsed" style="position: absolute; bottom: 15px; left:15px" id="affix-container-target">
```

Comment:
Inline styling is a bad practice, the minimum should be to use a CSS class instead and move the styles into `app.component.less`.

Absolute positioning is kind of an art form. It can be the fitting solution in some cases, but most of the time it's an escape hatch that just causes problems (for example when collapsing the menu, as you noticed), because it takes the element entirely out of the document flow. If the intent is simply to have this on the bottom of the menu, an easier solution would be

```
margin: auto 15px 15px 15px;
```

I haven't verified this here, but it should work (we use the same thing in Ververica Platform).

(The final nit would be that I'm guessing the value of 15px was more or less chosen at random, AntD actually uses a different progression for spacing, so technically 12px or 16px would be in line with the design language.)

Reply:
The margin positioning didn't work:
![image](https://user-images.githubusercontent.com/89049/122168648-d6cda500-ce7c-11eb-99fb-61eb5ae69081.png)

(But let's maybe address this once we have agreement where to place the download button in general)

Reply:
> But let's maybe address this once we have agreement where to place the download button in general

Sounds good. If it does stay here, I'd also not hard-oppose the absolute positioning, but alternatively we could also do this in pairing.